### PR TITLE
refactor(app): Add WPA2 personal auth modal to new connection UI

### LINKED ISSUE #2507: Manage Robot Connection: Connect to WPA2 Personal network
## overview

This ticket taking over for the WPA2 personal aspects of #2413.

As a run app user _with a WPA2 Personal Wi-Fi network_, I would like to be able to enter the appropriate authentication information (i.e. the password) to connect to my network.

## Acceptance Criteria
- [ ] After user selects a WPA2 personal network from the dropdown, connection modal is show
- [ ] Connection modal has password field
    - [ ] 'Show Password' checkbox below PW field
- [ ] User able to click 'Join' to attempt authentication
    - [ ] Form validation covered by #2416
- [ ] User able to click 'Cancel' to return to Robot Settings page

### Copy
- Title bar: "Wi-Fi network ${ssid} requires a WPA2 password"

### Designs
- https://zpl.io/Vqwz61m
    - Note: Not implementing "Remember this network" option because it is not technically feasible at this time

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR closes #2507. See ticket for background and acceptance criteria. Classified as a `refactor` because behavior is behind a feature flag.

## changelog

- refactor(app): Add WPA2 personal auth modal to new connection UI

## review requests

0. Run `make install-js` if you haven't since #2512 was merged
1. Launch app with new connection card enabled

    ```shell
    make -C app dev OT_APP_DEV_INTERNAL__MANAGE_ROBOT_CONNECTION__NEW_CARD=1
    ```

2. Select a WPA2 Personal network in the connection card dropdown

- [ ] Connect modal pops up
- [ ] Can enter password
    - [ ] Can click "Password:" label to activate field
- [ ] Can toggle "Show password"
    - [ ] Can click "Show password" label to activate field
- [ ] Can click "Cancel" to close modal without attempting join
- [ ] Can click overlay to close modal without attempting join
- [ ] Can click "Join" to join network
    - [ ] UI switches to spinner
    - [ ] Success modal pops up if join was successful
    - [ ] Error modal pops up if join was not successful

### CODE REVIEW

File: app/src/components/RobotSettings/SelectNetwork/index.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
+// @flow
```
Comment (mcous) [inline app/src/components/RobotSettings/SelectNetwork/index.js]: File renamed from `app/src/components/RobotSettings/SelectNetwork.js`, sorry for the diff noise

File: app/src/components/RobotSettings/SelectNetwork/ConnectForm.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import {InputField, CheckboxField} from '@opentrons/components'
+import {BottomButtonBar} from '../../modals'
+import FormTable, {FormTableRow} from './FormTable'
+
+import type {WifiConfigureRequest} from '../../../http-api-client'
+
+// TODO(mc, 2018-10-18): add 802.1x fields
+type AuthFieldName = 'psk'
+
+type AuthField = {
```
Comment (mcous) [inline app/src/components/RobotSettings/SelectNetwork/ConnectForm.js]: Note: this is the same field shape as returned by `GET /wifi/eap-options`

File: app/src/components/RobotSettings/SelectNetwork/ConnectForm.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    const {fields} = this.props
+
+    // TODO(mc, 2018-10-18): form validation
+    return (
+      <Formik
+        onSubmit={this.onSubmit}
+        render={({handleChange, handleSubmit, values}) => (
+          <form onSubmit={handleSubmit}>
+            <FormTable>
+              {fields.map(field => (
```
Comment (IanLondon) [inline app/src/components/RobotSettings/SelectNetwork/ConnectForm.js]: Nitpick (long-winded!): I think it should not be the server's responsibility to determine and specify pure UI concerns, so while the collection of wifi auth fields per endpoint is a data model concern that the server rightly should own, the order of those fields in the form (what is on the top, what is on the bottom) seems like a pure UI concern. If the design changes for the field ordering in Run App, the code change would be on the backend, which seems off to me. The `displayName` is also coming from the server, same problem.

Instead of an array of options with displayName, would it be cleaner if the backend sent out:

```js
{
  "name": "ttls/eap-md5",
  "options": {
    "identity": {
      "required": true,
      "type": "string"
    },
    "anonymousIdentity": {
      "required": false,
      "type": "string"
    },
    "caCert": {
      "required": false,
      "type": "file"
    },
    ...etc
```

And the Run App would have the responsibility to map the option keys to display names & to order them?

This is def not blocking, and not a big deal despite the block of text -- I just want to understand better why the backend is specifying displayNames and display order of the fields

Comment (mcous) [reply app/src/components/RobotSettings/SelectNetwork/ConnectForm.js]: I think this is a valid concern, the `/wifi/eap-options` endpoint as written does include UI stuff. I think in this case, though, there are definite benefits for the app to not actually know anything about the fields:

- Robot can be updated to support new auth types without _any_ front-end changes
- Backwards compatibility with older robot software is easily maintained
    - e.g. if we change the field ID of a given input type

Downsides:

- Display names are controlled by the server
    - This isn't ideal, but I also don't want to maintain this code in the app, away from where the fields are actually defined
- Order is controlled by the server
    - Honestly didn't even think about this, but yeah it's kinda weird
    - Same as above, though, I don't want to maintain a field ordering in the app

TL;DR: I _think_ I'd rather optimize for keeping auth field changes contained entirely to the API for now. Happy to talk about this more (and be convinced otherwise) in person while this is still behind a feature flag, though
